http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -528,9 +540,127 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
+      )
+    } yield ()
+
+  private def deleteDiskForApp(diskId: DiskId, appId: AppId)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val delete = for {
+      ctx <- ev.ask
+      _ <- handleDeleteDiskMessage(DeleteDiskMessage(diskId, Some(ctx.traceId)), true)
+    } yield ()
+
+    delete.onError {
+      case e =>
+        for {
+          ctx <- ev.ask
+          _ <- logger.error(e)(s"Failed to delete disk for app ${appId}")
+          _ <- appErrorQuery
+            .save(appId, KubernetesError(e.getMessage(), ctx.now, ErrorAction.DeleteGalaxyApp, ErrorSource.Disk, None))
+            .transaction
+        } yield ()
+    }
+  }
+
+  private def createDiskForApp(msg: CreateAppMessage)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val create = msg.createDisk match {
+      case true =>
+        for {
+          _ <- logger.info(s"Beginning disk creation for app ${msg.appId}")
+          ctx <- ev.ask
+          getAppOpt <- KubernetesServiceDbQueries.getActiveFullAppByName(msg.project, msg.appName).transaction
+          getApp <- F.fromOption(getAppOpt, AppNotFoundException(msg.project, msg.appName, ctx.traceId))
+          disk <- F.fromOption(
+            getApp.app.appResources.disk,
+            AppCreationException(
+              s"create disk was true for create app message, but app ${getApp.app.id} does not have a disk id saved"
+            )
+          )
+          _ <- handleCreateDiskMessage(CreateDiskMessage.fromDisk(disk, Some(ctx.traceId)), true)
+        } yield ()
+      case false => F.unit
+    }
+    create.onError {
+      case e =>
+        for {
+          ctx <- ev.ask
+          _ <- logger.error(e)(s"Failed to create disk for app ${msg.appId}")
+          _ <- appErrorQuery
+            .save(msg.appId,
+                  KubernetesError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Disk, None))
+            .transaction
+        } yield ()
+    }
+  }
+
+  def handleDeleteAppMessage(msg: DeleteAppMessage)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      _ <- logger.info(s"beginning delete app for app ${msg.project}/${msg.appName}")
+      initialGoogleCall = for {
+        getAppOpt <- KubernetesServiceDbQueries.getActiveFullAppByName(msg.project, msg.appName).transaction
+        getApp <- F.fromOption(getAppOpt, AppNotFoundException(msg.project, msg.appName, ctx.traceId))
+        deleteResult <- gkeInterp.deleteNodepool(getApp)
+      } yield (getApp, deleteResult)
+      (getApp, deleteResult) <- initialGoogleCall.onError {
+        case e =>
+          for {
+            ctx <- ev.ask
+            _ <- logger.error(e)(s"Failed to delete app for app ${msg.appId}")
+            _ <- appErrorQuery
+              .save(msg.appId,
+                    KubernetesError(e.getMessage(), ctx.now, ErrorAction.DeleteGalaxyApp, ErrorSource.Nodepool, None))
```

Comment:
same comment...`adaptError` here, and use `onError` on the whole fo comprehension. Also we put `ctx <- ev.ask` at the beginning of the for comp, so that it's available to all statements after, line 646 is duplicate